pints/_mcmc/_population.py

```diff
@@ -0,0 +1,269 @@
+#
+# Population MCMC
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+# Some code in this file was adapted from Myokit (see http://myokit.org)
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class PopulationMCMC(pints.SingleChainMCMC):
+    """
+    *Extends:* :class:`MCMC`
+
+    Creates a chain of samples from a target distribution, using the population
+    MCMC (simulated tempering) routine described in algorithm 1 in [1].
+
+    The algorithm goes through the following steps (after initialising ``N``
+    chains):
+
+    1. Mutation: randomly select chain ``i`` and update the chain using a
+    Markov kernel that admits ``pi_i`` as its invariant distribution.
+
+    2. Exchange: Select another chain ``j`` at random from the remaining and
+    swap the parameter vector of ``i`` and ``j`` with probability
+    ``min(1, A)``,
+
+    ``A = pi_i(x_j) * pi_j(x_i) / (pi_i(x_i) * pi_j(x_j))``
```

Comment:
change `pi_i` and `pi_j` to `p_i` and `p_j`

Reply:
I'm easy here. I'd prefer pi, personally as this is the naming convention used in the Holmes paper, but I'm not wed to it.

Reply:
Thing is, I read (bits of) the paper too, but looking at the code a few days later still read it as p-i-i, rather than pi-i (sir)

Reply:
I'd be happy with pi as long as you use something like a and b for indexing :)

Reply:
I've gone for p now, which matches with the code!